src/main/java/seedu/patientist/model/ward/Ward.java

```diff
@@ -110,6 +113,48 @@ public void deletePerson(Person person) {
         }
     }
 
+    /**
+     * Deletes a patient based on id from any ward. Patient must already exist in ward.
+     *
+     * @param idNumber The id number of the patient.
+     * @param patient The dummy patient used to check equality of {@code IdNumber}.
+     */
+    public void deletePatientById(IdNumber idNumber, Patient patient) {
+        requireAllNonNull(idNumber, patient);
+        if (!containsPatient(patient)) {
+            throw new PersonNotFoundException();
+        }
+        Iterator<Person> patientIterator = patients.iterator();
```

Comment:
I think there needs to be a check if the id in both is the same or make it such that it only excepts either id or patient. I think we should make it such that it accepts patient only because doesn't seem to be in use.